core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -442,14 +453,6 @@ public void acknowledgeSessionUpdate(String groupId, ShareRequestMetadata reqMet
         if (reqMetadata.epoch() == ShareRequestMetadata.INITIAL_EPOCH) {
             // ShareAcknowledge Request cannot have epoch as INITIAL_EPOCH (0)
             throw Errors.INVALID_SHARE_SESSION_EPOCH.exception();
-        } else if (reqMetadata.epoch() == ShareRequestMetadata.FINAL_EPOCH) {
```

Comment:
I don't think this is quite right. If I send in a ShareAcknowledge with FINAL_EPOCH, then handleShareAcknowledge will call this method with reqMetadata.epoch() == FINAL_EPOCH. Because this else-if has been removed, it's going to end up in the next else, and the epoch will not match the request epoch.

Reply:
Yeah, it should be corrected. Done.